buildSrc/src/main/java/com/google/firebase/gradle/plugins/FireEscapeArtifactPlugin.kt

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.api.publish.maven.plugins.MavenPublishPlugin
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.bundling.Zip
+
+class FireEscapeArtifactPlugin : Plugin<Project> {
+  var project: Project? = null
+
+  override fun apply(target: Project) {
+    project = target
+
+    project!!.afterEvaluate {
+      val firebaseLibrary = extensions.findByType(FirebaseLibraryExtension::class.java)
+      val fireEscapeTask =
+        tasks.create("mavenAarFireEscapeArtifact", Zip::class.java) { classifier = "fireescape" }
```

Comment:
Same as above, `create` is `reified`, and can be called like such:
```kotlin
val fireEscapeTask = tasks.create<Zip>("maven") {
 // ...
}
```

Additionally, `classifier` is deprecated on `Zip`, and you should instead use `archiveClassifier` like so:

```kotlin
archiveClassifier.set("fireescape")
```

So the finished product would be something like:
```kotlin
val fireEscapeTask = tasks.create<Zip>("maven") {
  archiveClassifier.set("fireescape")
}
```

Reply:
Attempting to use the reified syntax you described for create seems to make the compiler think I'm using the `name, actionTask` version of this function, and tells me no type arguments were expected, does something more need to be done?

Reply:
Double check that you're:
A.) Using the correct `Zip` class (as there seems to be two). The correct one is: `org.gradle.api.tasks.bundling.Zip`
B.) Have the DSL variant imported, which happens automatically on my machine but yours may differ. The package is `org.gradle.kotlin.dsl.create`